### PR TITLE
agent editor

### PR DESCRIPTION
**Pull request by Minilik94**

### CODE REVIEW

File: apps/web/src/lib/components/ui/agent-editor/create-agent.svelte [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			<div class="p-6">
+				<div class="flex w-full items-center gap-4">
+					<div class="mb-4 w-full">
+						<Label for="title" class="block text-sm font-medium ">Title</Label>
+						<Input
+							id="title"
+							name="title"
+							bind:value={$formAgent.title}
+							placeholder="Agent's title"
+							class="border-input placeholder:text-muted-foreground focus-visible:ring-ring col-span-3 mt-1  block  h-9 w-full  rounded-md border bg-transparent px-3 text-sm shadow-sm ring-offset-0 transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-offset-0 disabled:cursor-not-allowed disabled:opacity-50"
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/agent-editor/create-agent.svelte]: why are there so many styling classes 😅

File: apps/web/src/lib/components/ui/agent-editor/create-agent.svelte [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+						/>
+					</div>
+					<div class=" mb-4 mt-4 flex cursor-pointer items-center gap-4 p-2">
+						<Toggle
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/agent-editor/create-agent.svelte]: I know I said "toggle" in the chat. but you should be well versed in the shadcn components by now to know that the "switch" component is probably better for this UI. correct me if I'm wrong though

File: apps/web/src/lib/components/ui/agent-editor/create-agent.svelte [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+					col-span-3 mt-1  block h-24 w-full resize-none rounded-md border bg-transparent px-3 text-sm shadow-sm ring-offset-0 transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-offset-0 disabled:cursor-not-allowed disabled:opacity-50 [&::-webkit-scrollbar]:hidden"
+					></Textarea>
+				</div>
+				{#if $errors.description}
+					<p class="mb-2 text-red-500">{$errors.description}</p>
+				{/if}
+				<div class="flex w-full gap-4">
+					<div class="mb-4 flex w-full flex-col">
+						<Label for="models" class="text-on-background mb-2 font-semibold">Models</Label>
+						<select
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/agent-editor/create-agent.svelte]: Should this be a select component instead?
https://www.shadcn-svelte.com/docs/components/select

File: apps/web/src/lib/components/ui/agent-editor/edit-agent.svelte
Comment (antopiahk) [inline apps/web/src/lib/components/ui/agent-editor/edit-agent.svelte:1]: create a component to reuse some styling for edit-agent and create-agent

File: apps/web/src/lib/components/ui/library/agent-library.svelte [hunk 4]
```diff
@@ -93,8 +93,8 @@
 							Published
 						</DropdownMenu.CheckboxItem>
 						<DropdownMenu.CheckboxItem
-							checked={filterModel === 'gpt-3-turbo'}
-							on:click={() => updateFilterModel('gpt-3-turbo')}
+							checked={filterModel === 'gpt-3.5-turbo'}
+							on:click={() => updateFilterModel('gpt-3.5-turbo')}
 						>
 							GPT-3
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/library/agent-library.svelte]: make the actual text "GPT-3.5-turbo"

File: apps/web/src/lib/schema.ts [hunk 5]
```diff
@@ -1,33 +1,51 @@
-import { z } from "zod";
+import { z } from 'zod';
 
 export const formSchema = z.object({
```
Comment (antopiahk) [inline apps/web/src/lib/schema.ts:3]: this should be named more accurately

File: apps/web/src/routes/app/editor/agent/+page.server.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+		const id = url.searchParams.get('id');
+
+		const form = await superValidate(request, agentFormSchema);
+
+		console.log(form, 'from from agent');
+		if (!form.valid) {
+			return fail(400, { form, message: 'Could not edit agent' });
+		}
+
+		try {
```
Comment (antopiahk) [inline apps/web/src/routes/app/editor/agent/+page.server.ts:79]: @eksno is try-catch good practice?

File: apps/web/src/routes/app/editor/agent/+page.svelte [hunk 7]
```diff
@@ -1,5 +1,69 @@
...
+	import { CreateAgent, EditAgent } from '$lib/components/ui/agent-editor/';
+	import type { Agent } from '$lib/types/models';
+	import { Button } from '$lib/components/ui/button';
+	import { afterUpdate } from 'svelte';
+	import { invalidateAll } from '$app/navigation';
+
+	export let data;
+	export let form;
+
+	$: myAgents = data.getCurrentUserAgents.data;
```
Comment (antopiahk) [inline apps/web/src/routes/app/editor/agent/+page.svelte]: "get" only used for function names

**Review by antopiahk — COMMENTED** [review]
waiting on jonas' response on the try-catch issue

File: apps/web/src/lib/schema.ts (hunk 5 above)
Comment (eksno) [reply apps/web/src/lib/schema.ts:3]: wdym it should be named more accurately? It's a line importing zod

File: apps/web/src/routes/app/editor/agent/+page.server.ts (hunk 6 above)
Comment (eksno) [reply apps/web/src/routes/app/editor/agent/+page.server.ts:79]: It is not, only the line at risk of erroring should be in a try catch. Also on another note why are we returning "fail" with the status code "200" indicating success? this looks very weird.

**Pull request merged**